dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -50,7 +50,30 @@
 ]
 
 
-def _compute_res_dtype(*arrays, sycl_queue, dtype=None, out=None, casting="no"):
+def _call_syrk(x1, x2):
+    """
+    Check to see if `syrk` can be called instead of `gemm`.
+
+    It is assumed that x1 and x2 are usm_ndarray objects. These arrays have
+    already been validated to be 2-dimensional and contiguous. Therefore, this
+    function only verifies the following: Both arrays reference the same
+    memory. The number of rows in x1 equals the number of columns in x2. If one
+    array is C-contiguous, the other must be F-contiguous.
+
+    """
+    call_syrk = False
+    if (
```

Comment:
I wonder if that is correct. We are passing only `x1` to `bi._syrk()`. We have to check that `x2` is exact transpose of `x1`, isn't it?
Do we need to check that `x1.shape[0] == x2.shape[1] and x1.shape[1] == x2.shape[0]`?

Reply:
Thanks, it's clear now.